test/e2e/performanceprofile/functests/2_performance_update/updating_profile.go

```diff
@@ -561,7 +581,27 @@ var _ = Describe("[rfe_id:28761][performance] Updating parameters in performance
 
 			// revert node label to have the expected value
 			nodeLabel = testutils.NodeSelectorLabels
+
+			if mcpSelectorRemoved {
+				By("Restoring Machine config selector")
+				profile.Spec.MachineConfigPoolSelector = existingMcpSelector
+				spec, err := json.Marshal(profile.Spec)
+				Expect(err).ToNot(HaveOccurred())
```

Comment:
this works, but I think we should prepare the revert _before_ to commit the change. This way we maximize the steps we prepared, and we minimize the possible failures in the restore path.
IOW:
1. fetch the old mcp selector
2. JSON marshal it - prepate the data for the restoration
3. only now, only if *both* the above succeeded, change the cluster MCP selector.

This way in the restore path the only thing that can fail is the patch operation, but we can't avoid, it's the minimum we need. Applies also to the label handling.

Reply:
I have updated the code as per the above requirements.